roles/network_plugin/weave/tasks/pre-upgrade.yml

```diff
@@ -37,3 +37,5 @@
 
 - name: Weave pre-upgrade | Purge legacy weave docker containers
   shell: "docker ps -af 'name=^/weave.*' -q | xargs --no-run-if-empty docker rm -f"
+  retries: 3
```

Comment:
Ugh...well, then we should probably think about going away from devicemapper :D I made very good experience with overlay and overlay2 (requires a newer kernel).